src/coreclr/jit/gentree.cpp

```diff
@@ -19326,7 +19326,16 @@ FieldSeq* FieldSeqStore::Append(FieldSeq* a, FieldSeq* b)
         return a;
     }
 
-    assert(!"Duplicate field sequences!");
+    // In UB-like code (such as manual IL) we can see an addition of two static field addresses.
+    // Treat that as cancelling out the sequence, since the result will point nowhere.
+    //
+    // It may be possible for the JIT to encounter other types of UB additions, such as due to
+    // complex optimizations, inlining, etc. In release we'll still do the right thing by returning
+    // nullptr here, but the more conservative assert can help avoid JIT bugs
+
+    assert(a->GetKind() == FieldSeq::FieldKind::SimpleStaticKnownAddress);
+    assert(b->GetKind() == FieldSeq::FieldKind::SimpleStaticKnownAddress);
```

Comment:
Did you hit that assert?

Reply:
Yes, its what was discussed further up in https://github.com/dotnet/runtime/pull/97901#discussion_r1477066382

Basically, we have an IL test that explicitly adds two `ldsflda`. It does so after a `conv.i4`, but that is a `nop` on 32-bit and a user can write such IL without the `conv.i4` as well, so the assert was already somewhat incorrect just not hit due to us not constant folding this early.

Reply:
Ok. Yeah I think we discussed it with SingleAccretion at some point